### PR TITLE
[MERGE] stock: pass tests without demodata

### PR DESCRIPTION
**Pull request by hch-odoo**
Do not require the demo data to be able to run (and pass) the tests of stock.

task-2024202

### CODE REVIEW

File: addons/stock/tests/test_move.py [hunk 1]
```diff
@@ -930,13 +955,14 @@ def test_putaway_6(self):
...
         })
+        self.product_cable_management_box.categ_id = self.product_category_5.id
 
         # creation
         move1 = self.env['stock.move'].create({
             'name': 'test_putaway_6',
             'location_id': self.supplier_location.id,
             'location_dest_id': self.stock_location.id,
-            'product_id': self.env.ref('stock.product_cable_management_box').id,
+            'product_id': self.product_cable_management_box.id,
```
Comment (sle-odoo) [inline addons/stock/tests/test_move.py]: al these tests would work as well with self.product
there's not need to introduce cls.product_cable_management_box cls.product_category_5 cls.product_product_5 cls.uom_kg..

File: addons/stock/tests/test_move2.py [hunk 2]
```diff
@@ -1890,6 +1890,18 @@ def setUp(self):
...
+            'supplier': True,
+            'customer': False,
+            'is_company': True,
+            'street': '1164 Cambridge Drive',
+            'city': 'White Tanks',
+            'zip': 85340,
+            'email': 'wood.corner26@example.com',
+            'phone': '(623)-853-7197',
+            'website': 'http://www.wood-corner.com'
+            })
```
Comment (sle-odoo) [inline addons/stock/tests/test_move2.py]: self.partner = self.env['res.partner'].create({'name': 'partner})
keep itsimple

Comment (sle-odoo) [inline addons/stock/tests/test_move2.py]: please you should configure your linter

``` python
x = {
    'cc': 'cc',
    }
```
should be 
``` python
x = {
    'cc': 'cc',
}
```

File: addons/stock/tests/test_packing_neg.py [hunk 3]
```diff
@@ -7,6 +7,29 @@
...
+            'customer': False,
+            'supplier': True,
+            'is_company': True,
+            'street': '92 Emily Drive',
+            'city': 'Columbia',
+            'zip': 29201,
+            'email': 'ready.mat28@example.com',
+            'phone': '(803)-873-6126',
+            'website': 'http://www.ready-mat.com'
+            })
```
Comment (sle-odoo) [inline addons/stock/tests/test_packing_neg.py]: no need to store on self + keep it simple + indent the dict correctly

File: addons/stock/tests/test_proc_rule.py [hunk 4]
```diff
@@ -9,6 +9,35 @@
...
+        self.res_partner_2 = self.env['res.partner'].create({
+            'name': 'Deco Addict',
+            'is_company': True,
+            'street': '325 Elsie Drive',
+            'city': 'Franklin',
+            'zip': 26807,
+            'email': 'deco.addict82@example.com',
+            'phone': '(603)-996-3829',
+            'website': 'http://www.deco-addict.com'
+            })
```
Comment (sle-odoo) [inline addons/stock/tests/test_proc_rule.py]: keep it simple please

File: addons/stock/tests/test_quant.py [hunk 5]
```diff
@@ -590,7 +613,7 @@ def test_in_date_4b(self):
             'quantity': 1.0,
         })
 
-        in_date_location2 = datetime.now()
+        in_date_location2 = datetime.now() + timedelta(seconds=10)
```
Comment (sle-odoo) [inline addons/stock/tests/test_quant.py]: we didn't have any issue with this test untilnow, not sure it is needed

File: addons/stock/tests/test_quant.py [hunk 6]
```diff
@@ -579,8 +592,18 @@ def test_in_date_4(self):
...
+            'posx': 0,
+            'barcode': 2341892,
+            'location_id': self.stock_location.id
+            })
+        stock_location2 = self.env['stock.location'].create({
+            'name': 'Shelf 2',
+            'posx': 0,
+            'barcode': 3451985,
+            'location_id': self.stock_location.id
+            })
```
Comment (sle-odoo) [inline addons/stock/tests/test_quant.py]: indent correctly + the barcode and pos stuff isn't needed

File: addons/stock/tests/test_quant.py [hunk 7]
```diff
@@ -47,7 +47,15 @@ def setUpClass(cls):
...
-        cls.stock_location = cls.env.ref('stock.stock_location_stock')
+        cls.stock_location = cls.env['stock.location'].create({
+            'name': 'stock_location',
+            'usage': 'internal',
+        })
+        cls.env['stock.location'].create({
+            'name': 'subloc2',
+            'usage': 'internal',
+            'location_id': cls.stock_location.id,
+        })
```
Comment (sle-odoo) [inline addons/stock/tests/test_quant.py]: you don't even assign this location to a variable?

File: addons/stock/tests/test_inventory.py [hunk 8]
```diff
@@ -315,6 +315,7 @@ def test_inventory_8(self):
         """ Check inventory lines product quantity is 0 when inventory is
         started with `prefill_counted_quantity` disable.
         """
+        product = self.product1
```
Comment (sle-odoo) [inline addons/stock/tests/test_inventory.py]: useless

File: addons/stock/tests/test_move2.py [hunk 9]
```diff
@@ -1892,6 +1892,7 @@ def setUp(self):
             'categ_id': self.env.ref('product.product_category_all').id,
         })
         self.uom_unit = self.env.ref('uom.product_uom_unit')
+        self.res_partner_1 = self.env['res.partner'].create({'name': 'Partner'})
```
Comment (sle-odoo) [inline addons/stock/tests/test_move2.py]: self.partner

File: addons/stock/tests/test_proc_rule.py [hunk 10]
```diff
@@ -9,6 +9,16 @@
 
 class TestProcRule(TransactionCase):
 
+    def setUp(self):
+        super(TestProcRule, self).setUp()
+
+        self.uom_unit = self.env.ref('uom.product_uom_unit')
+        self.product_product_3 = self.env['product.product'].create({
```
Comment (sle-odoo) [inline addons/stock/tests/test_proc_rule.py]: self.product

File: addons/stock/tests/test_proc_rule.py [hunk 11]
```diff
@@ -9,6 +9,16 @@
...
+            'name': 'Desk Combination',
+            'type': 'consu',
+        })
+        self.res_partner_2 = self.env['res.partner'].create({'name': 'Partner'})
```
Comment (sle-odoo) [inline addons/stock/tests/test_proc_rule.py]: self.partner

File: addons/stock/tests/test_quant.py [hunk 12]
```diff
@@ -47,7 +47,15 @@ def setUpClass(cls):
             'type': 'product',
             'tracking': 'serial',
         })
-        cls.stock_location = cls.env.ref('stock.stock_location_stock')
+        cls.stock_location = cls.env['stock.location'].create({
+            'name': 'stock_location',
+            'usage': 'internal',
+        })
+        cls.stock_subloc2 = cls.env['stock.location'].create({
```
Comment (sle-odoo) [inline addons/stock/tests/test_quant.py]: i don't see where stock_subloc2 is used?

File: addons/stock/tests/test_stock_flow.py [hunk 13]
```diff
@@ -6,6 +6,17 @@
 from odoo.exceptions import UserError
 
 class TestStockFlow(TestStockCommon):
+    def setUp(cls):
+        super(TestStockFlow, cls).setUp()
+        cls.res_partner_company_1 = cls.env['res.partner'].create({
+            'name': 'My Company (Chicago)-demo',
+            'email': 'chicago@yourcompany.com',
+            })
+        cls.res_company_1 = cls.env['res.company'].create({
```
Comment (sle-odoo) [inline addons/stock/tests/test_stock_flow.py]: cls.company2 seems appropriate

File: addons/stock/tests/test_stock_flow.py [hunk 14]
```diff
@@ -6,6 +6,17 @@
 from odoo.exceptions import UserError
 
 class TestStockFlow(TestStockCommon):
+    def setUp(cls):
+        super(TestStockFlow, cls).setUp()
+        cls.res_partner_company_1 = cls.env['res.partner'].create({
```
Comment (sle-odoo) [inline addons/stock/tests/test_stock_flow.py]: cls.partner_company2

File: addons/stock/tests/test_stock_location_search.py [hunk 15]
```diff
@@ -7,7 +7,12 @@ class TestStockLocationSearch(common.TransactionCase):
     def setUp(self):
         super(TestStockLocationSearch, self).setUp()
         self.location = self.env['stock.location']
-        self.location_barcode = self.env.ref('stock.stock_location_14')
+        self.stock_location = self.env.ref('stock.stock_location_stock')
+        self.location_barcode = self.env['stock.location'].create({
+            'name': 'Shelf 2',
+            'barcode': 1201985,
+            'location_id': self.stock_location.id
+            })
```
Comment (sle-odoo) [inline addons/stock/tests/test_stock_location_search.py]: indent + before to rename in cls.sublocation for clarity sake

File: addons/stock/tests/test_warehouse.py [hunk 16]
```diff
@@ -6,6 +6,11 @@
 
 
 class TestWarehouse(TestStockCommon):
+
+    def setUp(self):
+        super(TestWarehouse, self).setUp()
+        self.res_partner_2 = self.env['res.partner'].create({'name': 'Deco Addict'})
```
Comment (sle-odoo) [inline addons/stock/tests/test_warehouse.py]: self.partner

File: addons/stock/tests/test_wise_operator.py [hunk 17]
```diff
@@ -17,10 +17,26 @@ def test_wise_operator(self):
             'uom_po_id': self.ref('uom.product_uom_unit'),
         })
 
+        self.res_partner_2 = self.env['res.partner'].create({'name': 'Deco Addict'})
```
Comment (sle-odoo) [inline addons/stock/tests/test_wise_operator.py]: self.partner

File: addons/stock/tests/test_wise_operator.py [hunk 18]
```diff
@@ -17,10 +17,26 @@ def test_wise_operator(self):
...
+            'barcode': 1231985,
+            'location_id': warehouse.lot_stock_id.id
+            })
+        self.stock_location_components = self.env['stock.location'].create({
+            'name': 'Shelf 1',
+            'barcode': 1231892,
+            'location_id': warehouse.lot_stock_id.id
+            })
+
+        self.res_partner_4 = self.env['res.partner'].create({'name': 'Ready Mat'})
```
Comment (sle-odoo) [inline addons/stock/tests/test_wise_operator.py]: self.partner2

File: addons/stock/tests/test_wise_operator.py [hunk 19]
```diff
@@ -17,10 +17,26 @@ def test_wise_operator(self):
...
+        self.stock_location_shelf2 = self.env['stock.location'].create({
+            'name': 'Shelf 2',
+            'barcode': 1231985,
+            'location_id': warehouse.lot_stock_id.id
+            })
+        self.stock_location_components = self.env['stock.location'].create({
+            'name': 'Shelf 1',
+            'barcode': 1231892,
+            'location_id': warehouse.lot_stock_id.id
+            })
```
Comment (sle-odoo) [inline addons/stock/tests/test_wise_operator.py]: indent

File: addons/stock/tests/test_wise_operator.py [hunk 20]
```diff
@@ -17,10 +17,26 @@ def test_wise_operator(self):
...
 
+        self.res_partner_2 = self.env['res.partner'].create({'name': 'Deco Addict'})
+
+        warehouse = self.env['stock.warehouse'].search([('company_id', '=', self.env.company.id)], limit=1)
+        self.stock_location_shelf2 = self.env['stock.location'].create({
+            'name': 'Shelf 2',
+            'barcode': 1231985,
+            'location_id': warehouse.lot_stock_id.id
+            })
+        self.stock_location_components = self.env['stock.location'].create({
```
Comment (sle-odoo) [inline addons/stock/tests/test_wise_operator.py]: self.shelf1 is understandable

File: addons/stock/tests/test_wise_operator.py [hunk 21]
```diff
@@ -17,10 +17,26 @@ def test_wise_operator(self):
             'uom_po_id': self.ref('uom.product_uom_unit'),
         })
 
+        self.res_partner_2 = self.env['res.partner'].create({'name': 'Deco Addict'})
+
+        warehouse = self.env['stock.warehouse'].search([('company_id', '=', self.env.company.id)], limit=1)
+        self.stock_location_shelf2 = self.env['stock.location'].create({
```
Comment (sle-odoo) [inline addons/stock/tests/test_wise_operator.py]: self.shelf2

File: addons/stock/tests/test_quant.py (hunk 12 above)
Comment (hsh-odoo) [reply addons/stock/tests/test_quant.py]: @sle-odoo this [testcase](https://github.com/odoo/odoo/pull/34258/files/b5974fdfb23ebe17ffc17ac67799e219ff4c4752#diff-eef6aab50d6b1f825856390112d96c1dR227) is using the child location of the stock_location. so, that's way we have to create it.

**Comment by sle-odoo** [comment]
robodoo r+ rebase-merge

**Comment by sle-odoo** [comment]
robodoo retry

**Comment by sle-odoo** [comment]
robodoo retry

**Pull request closed without merging**